torch_geometric/data/remote_backend_utils.py

```diff
@@ -0,0 +1,70 @@
+# This file defines a set of utilities for remote backends (backends that are
+# characterize as Tuple[FeatureStore, GraphStore]).
+from typing import Tuple, Union
+
+from torch_geometric.data.feature_store import FeatureStore
+from torch_geometric.data.graph_store import GraphStore
+from torch_geometric.typing import EdgeType, NodeType
+
+
+def num_nodes(
+    feature_store: FeatureStore,
+    graph_store: GraphStore,
+    query: Union[NodeType, EdgeType],
+) -> Union[int, Tuple[int, int]]:
```

Comment:
I am not liking this union here too much. It looks like separating the interface into
```python
num_nodes(feature_store, graph_store, query: NodeType) -> int
```

```python
size(feature_store, graph_store, query: EdgeType) -> Tuple[int, int]
```
would make more sense?

Reply:
Sgtm